tests/repository/test_sync.py

```diff
@@ -0,0 +1,75 @@
+import logging
+import pytest
+
+from pubtools.pulplib import (
+    Repository,
+    YumRepository,
+    Task,
+    Distributor,
+    DetachedException,
+    SyncOptions,
+    TaskFailedException,
+)
+
+
+@pytest.fixture
+def fixture_sync_async_response(requests_mocker):
+    requests_mocker.post(
+        "https://pulp.example.com/pulp/api/v2/repositories/some-repo/actions/sync/",
+        [{"json": {"spawned_tasks": [{"task_id": "task1"}]}}],
+    )
+
+
+@pytest.fixture
+def fixture_search_task_response(requests_mocker):
+    requests_mocker.post(
+        "https://pulp.example.com/pulp/api/v2/tasks/search/",
+        [{"json": [{"task_id": "task1", "state": "finished"}]}],
+    )
+
+
+def test_detached():
+    """publish raises if called on a detached repo"""
+    with pytest.raises(DetachedException):
+        Repository(id="some-repo").sync()
+
+
+def test_sync(
```

Comment:
The test name and doc string doesn't match at all what's happening in the test body... should this be named something like `test_sync_missing_feed`  with doc `sync fails if no feed is provided` ?

Reply:
yeah, makes sense